src/enclave/main.cpp

```diff
@@ -59,6 +61,13 @@ extern "C"
     }
 #endif
 
+    // Host and enclave versions must match. Otherwise the node may crash much
+    // later (e.g. unhandled ring buffer message on either end)
+    if (std::string(host_version) != ccf::ccf_version)
```

Comment:
I might be missing something here but the comment in the PR says you compare commits but this is comparing versions. Not that I think there is any issue with comparing versions.

Reply:
Good point. The `ccf::ccf_version` is retrieved from `common/version.h` which is generated by `cmake` based on the long version string which is derived by `git describe --tags`, and changes for every commit (`e.g. "ccf-1.0.0-rc1-40-ge1d549c04"` for my current local checkout). So yeah, it's referred to as version because it will be a nice and neat version for major releases (e.g. `ccf-1.0.0`) but is more specific for everything in between.